qiskit/algorithms/evolvers/real/trotterization/trotter_qrte.py

```diff
@@ -0,0 +1,213 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An algorithm to implement a Trotterization real time-evolution."""
+
+from typing import Union, Dict, Optional
+
+from qiskit.algorithms import EvolutionProblem, EvolutionResult, RealEvolver, eval_observables
+from qiskit.circuit import Parameter
+from qiskit.opflow import (
+    OperatorBase,
+    SummedOp,
+    PauliOp,
+    CircuitOp,
+    ExpectationBase,
+    CircuitSampler,
+)
+from qiskit.circuit.library import PauliEvolutionGate
+from qiskit.providers import Backend, BaseBackend
+from qiskit.synthesis import ProductFormula, LieTrotter
+from qiskit.utils import QuantumInstance, algorithm_globals
+from .trotter_ops_validator import is_op_bound
+
+
+class TrotterQrte(RealEvolver):
+    """ Class for performing Quantum Real Time Evolution using Trotterization.
+    Type of Trotterization is defined by a ProductFormula provided.
+
+    Examples:
+
+        .. jupyter-execute::
+
+            from qiskit.opflow import X, Y, Zero
+            from qiskit.algorithms import EvolutionProblem, EvolutionResult
+            from qiskit.algorithms.evolvers.real.implementations.\
+                trotterization.trotter_qrte import TrotterQrte
+
+            operator = X + Z
+            initial_state = Zero
+            time = 1
+            evolution_problem = EvolutionProblem(operator, 1, initial_state)
+            # LieTrotter with 1 rep
+            trotter_qrte = TrotterQrte(evolution_problem)
+            evolved_state = trotter_qrte.evolve().evolved_state
+    """
+
+    def __init__(
+        self,
+        product_formula: ProductFormula = None,
+        expectation: Optional[ExpectationBase] = None,
+        quantum_instance: Union[QuantumInstance, BaseBackend, Backend] = None,
+    ) -> None:
+        """
+        Args:
+            product_formula: A Lie-Trotter-Suzuki product formula. The default is the Lie-Trotter
+                first order product formula with a single repetition.
+            expectation: An instance of ExpectationBase which defines a method for calculating
+                expectation values of EvolutionProblem.aux_operators.
+            quantum_instance: A quantum instance used for calculations.
+        """
+        if product_formula is None:
+            product_formula = LieTrotter()
+        self._product_formula = product_formula
```

Comment:
You don't have to change this, but since the setters and getters for these properties don't include any logic we could also just make them public properties. That'll reduce the boilerplate code. 

Yes, the getters and setters won't show up in the docs, but as long as they are well-document in the initializer I would argue that's okay 🙂

Reply:
At present I see only getters for product formula and expectation (quantum instance has both getter and setter). Usually the algos we have are not only reusable but updateable too.